src/auth/auth.controller.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Controller,
+  HttpStatus,
+  Get,
+  Query,
+  HttpException,
+  Post,
+  UseGuards,
+  Req,
+} from '@nestjs/common';
+import { AuthService } from './auth.service';
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
+import { JwtAuthGuard } from './jwt-auth.guard';
+import { TRequest } from 'src/common/types/user.types';
+
+@ApiTags('Auth')
+@Controller()
+export class AuthController {
+  constructor(private readonly authService: AuthService) {}
+
+  @Get('oauth/callback')
+  async githubAuthCallback(@Query('code') code: string) {
+    try {
+      const jwtToken = await this.authService.handleGitHubCallback(code);
+
+      return jwtToken;
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
```

Comment:
PullProofAI

There is a potential issue in error handling logic in the `githubAuthCallback` method. The condition `if (error.status)` might not be sufficient to determine if an error status exists. It's recommended to validate the error object more thoroughly to ensure correct HTTP status codes are returned.

#### Suggestions

You should enhance the error handling logic to properly check for the presence of an error status and handle it accordingly. Consider revising the condition to ensure accurate HTTP status codes are returned based on the error.

```typescript
28:       if (error.status !== undefined) {
29:         throw new HttpException(error.message, error.status);
30:       } else {
31:         throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
32:       }
```


<!-- This is an auto-generated comment by PullProofAI -->